dynawo/sources/Modeler/Common/DYNSubModel.h

```diff
@@ -317,34 +317,29 @@ class SubModel {
   virtual void rotateBuffers() = 0;
 
   /**
-   * @brief get the index of variables used to define the jacobian associated to a calculated variable
+   * @brief get the indexes global of the variables used to compute a calculated variable
    *
    * @param iCalculatedVar index of the calculated variable
+   * @param indexes vector to fill with the indexes
    *
-   * @return index of variables used to define the jacobian
    */
-  virtual std::vector<int> getDefJCalculatedVarI(int iCalculatedVar) = 0;
+  virtual void getIndexesOfVariablesUsedForCalculatedVarI(unsigned iCalculatedVar, std::vector<int>& indexes) const = 0;
```

Comment:
What is the reason of directly modifying the vector references instead of returning them? I'm always a bit surprised to have a get method that is void.

Reply:
returning a vector (with no reference) hides a copy, and potentially a second one when we assign it the result to a local vector. So by passing the vector as a reference parameter we save  up to 2 copies per call.